### DISCUSSION TITLE
useAtom in Next.Js 13 (appdir)

### DISCUSSION THREAD

**Discussion by Tygore** [pmndrs/jotai#1848] [Q&A]
Can I use useAtom without "use client"; in page.tsx (Next.js 13 with appdir)? I'm trying to find a manual, but so far without success.

**Comment by dai-shi** [comment]
`useAtom` uses `useReducer` internally, which isn't allowed in server components.

https://github.com/reactjs/rfcs/blob/main/text/0188-server-components.md#capabilities--constraints-of-server-and-client-components
<img width="925" alt="image" src="https://user-images.githubusercontent.com/490574/226550441-f2dc4e95-4a5d-4438-a215-26cb9a65fce2.png">

**Reply by Tygore** [reply]
It's clear. Those. no options to use Jotai with RSC (get value from atom in RSC)?
For example Zustand I can use useStore.setState(...

p.s. I love using your projects. Thank you @dai-shi

**Reply by dai-shi** [reply]
Being stateless on servers mean you shouldn't keep state across requests.
So using Zustand's useStore.setState() is wrong in the first place.

Anyway, the same thing can be done with Jotai's store API.
```js
import { atom, createStore } from 'jotai';

const myAtom = atom('hello');
const myStore = createStore();

myStore.set(myAtom, 'hey');
```
